src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RapidModeStepRunner.java

```diff
@@ -94,28 +94,32 @@ public void runOnline(BlackDuckRunData blackDuckRunData, NameVersion projectVers
             }
         });
         
-        stepHelper.runToolIfIncluded(DetectTool.CONTAINER_SCAN, "Container Scanner", () -> {
-            logger.debug("Stateless container scan detected.");
-
-            // Check if this is an SCA environment.
-            if (scaaasFilePath.isPresent()) {
-                List<HttpUrl> containerResultUrls = new ArrayList<>();
-                invokeBdbaRapidScan(blackDuckRunData, projectVersion, blackDuckUrl, containerResultUrls, true, scaaasFilePath.get());
-                processScanResults(containerResultUrls, parsedUrls, formattedCodeLocations, DetectTool.CONTAINER_SCAN.name());
-            } else {
-                logger.debug("Determining if configuration is valid to run a container scan.");
-                ContainerScanStepRunner containerScanStepRunner = new ContainerScanStepRunner(operationRunner, projectVersion, blackDuckRunData, gson);
-                if (containerScanStepRunner.shouldRunContainerScan()) {
-                    logger.debug("Invoking stateless container scan.");
-                    UUID scanId = containerScanStepRunner.invokeContainerScanningWorkflow();
-                    String statelessScanEndpoint = operationRunner.getScanServicePostEndpoint();
-                    HttpUrl scanServiceUrlToPoll = new HttpUrl(blackDuckUrl + statelessScanEndpoint + "/" + scanId.toString());
-                    parsedUrls.add(scanServiceUrlToPoll);
+        stepHelper.runToolIfIncludedWithCallbacks(
```

Comment:
What is the difference here between `.runToolIfIncluded(..) `vs `.runToolIfIncludedWithCallbacks(..)`? Assuming it has to do with what is necessary for success/failure status reporting to work but if you could shed some light please

Reply:
For example I see we'd be missing info level logs such as 

> logger.info("Will include the " + name + " tool.");

Is that intentional?

Reply:
`runToolIfIncludedWithCallbacks` also calls the `runToolIfIncluded` method, but adds 2 callback methods to its signature for a success and failure event respectively. This comes into play with Detect's event system where the failure callback get triggered if anything inside the `() -> {...}` lambda part throws an exception, and the success callback gets triggered if everything in that lambda block went well. 
You can also take a look at #925 to see the definition and usage of these callbacks for intelligent scans.

> For example I see we'd be removing info level logs such as
> logger.info("Will include the " + name + " tool.");
> Is that intentional?

That logger info line is not getting removed since `runToolIfIncludedWithCallbacks` also calls the `runToolIfIncluded` method.